scripts/example_workflow_builder.py

```diff
@@ -0,0 +1,104 @@
+from random import Random
+
+from immutablecollections import immutableset
+from vistautils.parameters import Parameters
+from vistautils.parameters_only_entrypoint import parameters_only_entry_point
+
+from pegasus_wrapper.artifact import ValueArtifact
+from pegasus_wrapper.locator import Locator, _parse_parts
+from pegasus_wrapper.pegasus_utils import build_submit_script
+from pegasus_wrapper.resource_request import SlurmResourceRequest
+from pegasus_wrapper.workflow import WorkflowBuilder
+from scripts.multiply_by_x import main as multiply_by_x_main
+from scripts.sort_nums_in_file import main as sort_nums_main
+
+
+def example_workflow(params: Parameters):
+    """
+    Example Workflow Creator for individuals to create a simple DAX and submit
+    to Pegasus for testing & learning purposes.
+    """
+    tmp_path = params.creatable_directory("dir")
+
+    # Generating parameters for initializing a workflow
+    workflow_params = Parameters.from_mapping(
+        {
+            "workflow_name": "Test",
+            "workflow_created": "Testing",
+            "workflow_log_dir": str(tmp_path / "log"),
+            "workflow_directory": str(tmp_path / "working"),
+            "site": "saga",
+            "namespace": "test",
+            "partition": "scavenge",
+        }
+    )
+
+    # Basic slurm resource request params
+    slurm_params = Parameters.from_mapping(
+        {"partition": "scavenge", "num_cpus": 1, "num_gpus": 0, "memory": "1G"}
+    )
+
+    # Our source input for the sample jobs
+    multiply_input_file = tmp_path / "raw_nums.txt"
+
+    random = Random()
+    random.seed(0)
+    nums = immutableset(int(random.random() * 100) for _ in range(0, 25))
+
+    multiply_output_file = tmp_path / "multiplied_nums.txt"
+    sorted_output_file = tmp_path / "sorted_nums.txt"
+
+    # Write a list of numbers out to be able to run the workflow
+    with multiply_input_file.open("w") as mult_file:
+        mult_file.writelines(f"{num}\n" for num in nums)
+
+    # Params for the multiply job
+    multiply_params = Parameters.from_mapping(
+        {"input_file": multiply_input_file, "ouput_file": multiply_output_file, "x": 4}
+    )
+    # Params for the sort job
+    sort_params = Parameters.from_mapping(
+        {"input_file": multiply_output_file, "output_file": sorted_output_file}
+    )
+
+    resources = SlurmResourceRequest.from_parameters(slurm_params)
+    workflow_builder = WorkflowBuilder.from_params(workflow_params)
+
+    multiply_job_name = Locator(_parse_parts("jobs/multiply"))
+    multiply_artifact = ValueArtifact(
+        multiply_output_file,
+        depends_on=workflow_builder.run_python_on_parameters(
```

Comment:
I find in-lining e.g. `multiply_params` as a Python dict here (it will auto-convert) is more readable:

```python
workflow_builder.run_python_on_parameters(
   jobs_locator / multiply,
   multiply_by_x_main,
   {"input_file": multiply_input_file, "ouput_file": multiply_output_file, "x": 4},
depends_on=[])
```